### PR TITLE
✨Feature/request timeout

### PR DESCRIPTION
**Pull request by tyfoo**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ export default function(opts) {
         apiClient,
         concurrency,
 
-        processJob: requestProcessor(),
+        processJob: requestProcessor(opts),
```
Comment (KATT) [inline src/index.js]: :fire:

File: src/lib/request-processor.js [hunk 2]
```diff
@@ -4,11 +4,12 @@ import HTMLEntities from 'he';
 
 const debug = require('debug')('dice:timeturner:request-processor');
 
-export default function(opts) {
+export default function(opts = {}) {
```
Comment (KATT) [inline src/lib/request-processor.js]: :fire:

File: src/schemas/Request.js [hunk 3]
```diff
@@ -9,14 +9,15 @@ const SUPPORTED_HTTP_METHODS = ['GET', 'POST', 'PUT', 'DELETE', 'PATCH'];
...
-    error : {type: Schema.Types.Mixed, default: null},
+    url    : {type: String, required: true, trim: true },
+    date   : {type: Date, required: true },
+    method : {type: String, enum: SUPPORTED_HTTP_METHODS, required: true },
+    body   : {type: Schema.Types.Mixed },
+    state  : {type: String, enum: STATES, required: true, default: 'SCHEDULED' },
+    job_id : {type: String},
+    notes  : {type: Schema.Types.Mixed},
+    error  : {type: Schema.Types.Mixed, default: null},
+    timeout: {type: Number},
```
Comment (KATT) [inline src/schemas/Request.js]: needs a `default` - align with [request](https://github.com/request/request)?

Comment (tyfoo) [inline src/schemas/Request.js]: I was thinking default should be `undefined` so it would just use `request` default?

**Comment by KATT** [comment]
Leaving this for now. If we implement this it needs to be done properly

* errors for each attempt needs to be logged
* introspection through some special state so you can see what's running & when their next attempt is
* it shouldn't block the queue, rather getting triggered again when the next attempt is due
* maybe cancellation through API-call

**Pull request closed without merging**